lib/worker.ex

```diff
@@ -3,43 +3,60 @@ defmodule Monarch.Worker do
   Defines the Monarch Worker for running Oban jobs.
   """
 
-  use Oban.Worker
+  use Oban.Worker,
+    unique: [
+      fields: [:args, :worker],
+      keys: [:job, :repo]
+    ]
 
+  @impl Oban.Worker
   def perform(job) do
     worker = String.to_existing_atom(job.args["job"])
     repo = String.to_existing_atom(job.args["repo"])
 
     # Wraps a chunk of records to be updated in a transaction
     {:ok, action} =
-      if apply(worker, :skip, []) do
-        {:ok, :halt}
-      else
-        repo.transaction(fn ->
-          case apply(worker, :query, []) do
-            [] ->
-              :halt
-
-            chunk ->
-              apply(worker, :update, [chunk])
-              :cont
-          end
-        end)
+      cond do
+        apply(worker, :skip, []) ->
+          {:ok, :halt}
+
+        Monarch.completed?(repo, worker) ->
```

Comment:
Fixes a race condition:

1) Job one starts and crashes due to timeout, retries in 5s
2) Job two gets enqueued because job one is running
3) Oban automatically restarts job one
4) Job one and Job two are running in parallel, doing the same thing, they keep crash looping
5) Eventually, one of the jobs completes and writes a completion log to the db
6) The other job is still running and continues crash looping until it runs out of retries (takes a few days..)

Multiple that by running new jobs on every boot = slowly bleeding to death due to job restarts and random queries being made.

This means before trying to execute _anything_, make sure the job hadn't already succeeded since the last run.